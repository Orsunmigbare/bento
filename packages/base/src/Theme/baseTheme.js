import { ThemeManager } from "@staccx/theme"

export const theme = {
  name: "Default theme",
  spacing: {
    huge: "96px",
    large: "48px",
    mediumPlus: "32px",
    medium: "24px",
    small: "12px",
    tiny: "6px",
    micro: "3px"
  },
  wrapper: {
    small: "420px",
    medium: "640px",
    large: "1024px",
    breakout: "640px"
  },
  color: {
    white: "#fff",
    bg: "#fff",
    subtleHover: "#F7F7FF",
    bgGray: "#f7f7ff",
    line: "#EBEBF2",
    gray: "#767676",
    wcag: "#777",
    disabled: "#BABABA",
    primary: "#00a190",
    secondary: "#F5A623",
    text: "#2b2b2b",
    negative: "#ea526f",
    positive: "#4A90E2",
    warning: "#c54059",
    black: "#000",
    green: "#00a190" // for toggle eks
  },
  fontFamily: {
    heading: "sans-serif",
    body: "sans-serif"
  },
  font: {
    h1: "36px",
    h2: "26px",
    h3: "22px",
    h4: "18px",
    h5: "16px",
    h6: "14px",
    base: "14px",
    tiny: "12px",
    input: "16px",
    huge: "44px"
  },
  targetSize: {
    small: "36px",
    normal: "44px"
  },
  borderRadius: "2px"
}

const manager = new ThemeManager(theme)
export default manager
