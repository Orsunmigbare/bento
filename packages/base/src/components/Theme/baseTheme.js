export const theme = {
  name: "StaccXBase",
  spacing: {
    huge: "96px",
    large: "48px",
    mediumPlus: "32px",
    medium: "24px",
    small: "12px",
    tiny: "6px"
  },
  wrapper: {
    small: "420px",
    medium: "640px",
    large: "1024px"
  },
  color: {
    bg: "#fff",
    bgGray: "#f7f7ff",
    primary: "#00a190",
    gray: "#767676",
    text: "#2b2b2b",
    white: "#fff",
    red: "#ea526f",
    green: "#00a190",
    blue: "#988EFB",
    yellow: "#f4da87",
    line: "#EBEBF2",
    wcag: "#777",
    disabled: "#BABABA",
    warning: "#c54059",
    black: "#000",
    subtleHover: "#F7F7FF"
  },
  font: {
    size: {
      h1: "36px",
      h2: "26px",
      h3: "22px",
      h4: "18px",
      h5: "16px",
      h6: "14px",
      base: "14px",
      tiny: "12px",
      input: "16px",
      huge: "44px"
    },
    heading: "Libre Baskerville",
    body: "Libre Franklin"
  },
  targetSize: {
    small: "36px",
    normal: "44px"
  },
  borderRadius: "2px"
}

export default theme
