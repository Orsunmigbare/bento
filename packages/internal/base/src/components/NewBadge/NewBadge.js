import React from "react"
import styled from "styled-components"
import PropTypes from "prop-types"
import { color, fontWeight, themify, themeProps } from "@staccx/theme"
import fontSmoothing from "../../Styles/fontsmoothing"
import themePropTypes from "../constants/themePropTypes"

const NewBadge = ({ number, children }) => (
  <NewBadgeWrapper>
    {children}
    {number > 0 && (
      <Badge number={number}>
        <span>{number}</span>
      </Badge>
    )}
  </NewBadgeWrapper>
)

NewBadge.themeProps = {
  wrapper: {
    name: "NEW_BADGE_WRAPPER",
    description: "",
    type: themePropTypes.style
  },
  badge: {
    name: "NEW_BADGE_BADGE",
    description: "",
    type: themePropTypes.style
  }
}

const NewBadgeWrapper = styled.div`
  position: relative;
  display: inline-block;
  ${themify(NewBadge.themeProps.wrapper)};
`

const Badge = styled.span`
  display: flex;
  justify-content: center;
  align-items: center;
  width: 20px;
  height: 20px;
  line-height: 1;
  background: ${color.warning};
  border-radius: 50%;
  text-align: center;
  color: ${color.white};
  font-size: ${p => (p.number < 99 ? "12px" : "10px")};
  font-weight: ${fontWeight.bold};
  right: -12px;
  bottom: -12px;
  position: absolute;
  ${fontSmoothing};
  ${themify(NewBadge.themeProps.badge)};
`

NewBadge.propTypes = {
  number: PropTypes.number.isRequired,
  children: themeProps.children
}

export default NewBadge
