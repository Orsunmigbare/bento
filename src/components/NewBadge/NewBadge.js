import React from "react"
import styled from "styled-components"
import PropTypes from "prop-types"
import fontSmoothing from "../../theming/styles/fontsmoothing"
import { color, fontWeight, applyVariants } from "../../theming"
import { commonPropTypes, themePropTypes } from "../../constants/themeContants"

const NewBadge = ({ number, children, variant }) => (
  <NewBadgeWrapper variant={variant}>
    {children}
    {number > 0 && (
      <Badge number={number} variant={variant}>
        <span>{number}</span>
      </Badge>
    )}
  </NewBadgeWrapper>
)

NewBadge.themeProps = {
  wrapper: {
    name: "NEW_BADGE_WRAPPER",
    description: "",
    type: themePropTypes.style
  },
  badge: {
    name: "NEW_BADGE_BADGE",
    description: "",
    type: themePropTypes.style
  }
}

const NewBadgeWrapper = styled.div`
  position: relative;
  display: inline-block;
  ${applyVariants(NewBadge.themeProps.wrapper)};
`

const Badge = styled.span`
  display: flex;
  justify-content: center;
  align-items: center;
  width: 20px;
  height: 20px;
  line-height: 1;
  background: ${color.warning};
  border-radius: 50%;
  text-align: center;
  color: ${color.white};
  font-size: ${p => (p.number < 99 ? "12px" : "10px")};
  font-weight: ${fontWeight.bold};
  right: -12px;
  bottom: -12px;
  position: absolute;
  ${fontSmoothing};
  ${applyVariants(NewBadge.themeProps.badge)};
`

NewBadge.propTypes = {
  /**
   * Number will be displayed on the badge
   */
  number: PropTypes.number.isRequired,
  children: commonPropTypes.children
}

export default NewBadge
