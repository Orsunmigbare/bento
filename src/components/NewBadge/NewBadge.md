```jsx
<NewBadge number={900}>
  <Caret />
</NewBadge>
```
