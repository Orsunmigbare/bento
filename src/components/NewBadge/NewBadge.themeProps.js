import { themePropTypes, createVariantsFunctionFactory } from "../../theming"

const themeProps = {
  wrapper: {
    name: "NEW_BADGE_WRAPPER",
    description: "",
    type: themePropTypes.style
  },
  badge: {
    name: "NEW_BADGE_BADGE",
    description: "",
    type: themePropTypes.style
  }
}

export default createVariantsFunctionFactory(themeProps)
