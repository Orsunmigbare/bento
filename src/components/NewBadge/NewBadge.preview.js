import React from "react"
import NewBadge from "./NewBadge"
import Label from "../Forms/Label/Label"
import Layout from "../Layout/Layout/Layout"
import CurrencyInput from "../Forms/Input/CurrencyInput/CurrencyInput"
const preview = {
  title: "Badge",
  category: "Components/Layout",
  tags: ["layout"],
  component: NewBadge,
  render: props => <NewBadge number={props.number || 1} />,
  renderExampleController: ({ setComponentState }) => (
    <Layout grid="columns">
      <div>
        <Label for="reverseRadio">reverse</Label>
        <CurrencyInput
          id={"badgeValue"}
          onChange={e => setComponentState({ number: e.target.value })}
        />
      </div>
    </Layout>
  )
}

export default preview
