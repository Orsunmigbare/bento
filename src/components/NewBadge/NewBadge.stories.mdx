import { Meta, Story, Preview, Props } from '@storybook/addon-docs/blocks';
import NewBadge from "./NewBadge"

<Meta title="Components|NewBadge" component={NewBadge}/>

# NewBadge
Badge are used to add additional information to content

```javascript
import { NewBadge } from "@staccx/bento"
```

## Standard NewBadge
Badge takes a number prop and will display that number in the badge

<Preview>
  <Story name="Standard">
   <NewBadge number={11} />
  </Story>
</Preview>



## Long Badge
Badges are not made for too long numbers

<Preview>
  <Story name="Long Badge">
   <NewBadge number={10000} />
  </Story>
</Preview>

## Badge with children
A badge can also take children elements

<Preview>
  <Story name="Children">
   <NewBadge number={10}>Messages</NewBadge>
  </Story>
</Preview>

## Props for NewBadge
<Props of={NewBadge} />
