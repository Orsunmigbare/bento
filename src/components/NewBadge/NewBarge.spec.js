import React from "react"
import renderer from "react-test-renderer"
import { ThemeProvider } from "styled-components"
import baseTheme from "../../theming/themes/baseTheme"
import NewBadge from "./NewBadge"

describe("NewBadge", () => {
  describe("Snapshots", () => {
    it("No props", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <NewBadge />
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("number", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <NewBadge number={1000} />
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("with children", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <NewBadge number={1000}>test</NewBadge>
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("number", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <NewBadge number={10} />
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})
