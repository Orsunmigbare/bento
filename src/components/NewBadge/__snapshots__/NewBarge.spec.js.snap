// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewBadge Snapshots No props 1`] = `
<div
  className="sc-AxjAm jyWUIT"
/>
`;

exports[`NewBadge Snapshots number 1`] = `
<div
  className="sc-AxjAm jyWUIT"
>
  <span
    className="sc-AxirZ hHUunu"
  >
    <span>
      1000
    </span>
  </span>
</div>
`;

exports[`NewBadge Snapshots number 2`] = `
<div
  className="sc-AxjAm jyWUIT"
>
  <span
    className="sc-AxirZ kZsFQc"
  >
    <span>
      10
    </span>
  </span>
</div>
`;

exports[`NewBadge Snapshots with children 1`] = `
<div
  className="sc-AxjAm jyWUIT"
>
  test
  <span
    className="sc-AxirZ hHUunu"
  >
    <span>
      1000
    </span>
  </span>
</div>
`;
